configure.py/cmake : build failure when passing multiple linker flags via --ldflags

Build fails when running configure with the following command :

```
  ./configure.py --mode=dev --use-cmake --cflags='-g -gz' --ldflags='-g -gz'
```
Error :

```
-- Check for working C compiler: /usr/lib64/ccache/clang - broken
CMake Error at /usr/share/cmake/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/usr/lib64/ccache/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/home/Scylla/scylladb/build/CMakeFiles/CMakeScratch/TryCompile-uAEY3k'
    
    Run Build Command(s): /usr/bin/ninja-build -v cmTC_3102f
    [1/2 : 0.030s] ccache /usr/lib64/ccache/clang -DCMAKE_INTDIR=\"Debug\"  -g -MD -MT CMakeFiles/cmTC_3102f.dir/Debug/testCCompiler.c.o -MF CMakeFiles/cmTC_3102f.dir/Debug/testCCompiler.c.o.d -o CMakeFiles/cmTC_3102f.dir/Debug/testCCompiler.c.o -c /home/Scylla/scylladb/build/CMakeFiles/CMakeScratch/TryCompile-uAEY3k/testCCompiler.c
    [2/2 : 0.043s] : && /usr/lib64/ccache/clang -g -g;-gz CMakeFiles/cmTC_3102f.dir/Debug/testCCompiler.c.o -o Debug/cmTC_3102f   && :
    FAILED: Debug/cmTC_3102f 
    : && /usr/lib64/ccache/clang -g -g;-gz CMakeFiles/cmTC_3102f.dir/Debug/testCCompiler.c.o -o Debug/cmTC_3102f   && :
    clang: error: no input files
    /bin/sh: line 1: -gz: command not found
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:3 (project)


-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/home/Scylla/scylladb/./configure.py", line 2976, in <module>
    configure_using_cmake(args)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "/home/Scylla/scylladb/./configure.py", line 2969, in configure_using_cmake
    subprocess.check_call(cmake_command, shell=False, cwd=source_dir)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '-DCMAKE_CONFIGURATION_TYPES=Dev', '-DCMAKE_CROSS_CONFIGS=Dev', '-DCMAKE_DEFAULT_CONFIGS=Dev', '-DCMAKE_C_COMPILER=clang', '-DCMAKE_CXX_COMPILER=clang++', '-DCMAKE_CXX_FLAGS=-g -gz -O0', '-DCMAKE_EXE_LINKER_FLAGS=-g;-gz', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DScylla_CHECK_HEADERS=ON', '-DScylla_DIST=OFF', '-DScylla_TEST_TIMEOUT=7200', '-DScylla_TEST_REPEAT=1', '-DScylla_ENABLE_LTO=ON', '-DScylla_PROFDATA_COMPRESSED_FILE=pgo/profiles/x86_64/profile.profdata.xz', '-G', 'Ninja Multi-Config', '-B', '/home/Scylla/scylladb/build', '-S', '/home/Scylla/scylladb']' returned non-zero exit status 1.

```

--- a/configure.py
+++ b/configure.py
@@ -2920,7 +2920,7 @@ def configure_using_cmake(args):
         'CMAKE_C_COMPILER': args.cc,
         'CMAKE_CXX_COMPILER': args.cxx,
         'CMAKE_CXX_FLAGS': args.user_cflags,
-        'CMAKE_EXE_LINKER_FLAGS': semicolon_separated(args.user_ldflags),
+        'CMAKE_EXE_LINKER_FLAGS': args.user_ldflags,
         'CMAKE_EXPORT_COMPILE_COMMANDS': 'ON',
         'Scylla_CHECK_HEADERS': 'ON',
         'Scylla_DIST': 'ON' if args.enable_dist in (None, True) else 'OFF',

--- a/configure.py
+++ b/configure.py
@@ -1808,7 +1808,7 @@ for mode in modes:
     # Those flags are passed not only to Scylla objects, but also to libraries
     # that we compile ourselves.
     modes[mode]['lib_cflags'] = user_cflags
-    modes[mode]['lib_ldflags'] = user_ldflags + linker_flags
+    modes[mode]['lib_ldflags'] = user_ldflags + ' ' + linker_flags
 
 
 def prepare_advanced_optimizations(*, modes, build_modes, args):

